source/2_platform/rex_opengl/src/resources/buffer.cpp

```diff
@@ -35,41 +34,43 @@ namespace rex
   {
     REX_ASSERT(*this != other);
 
-    this->m_resource_id = rsl::exchange(other.m_resource_id, 0);
+    this->m_resource_id     = rsl::exchange(other.m_resource_id, 0);
     this->m_resource_target = rsl::exchange(other.m_resource_target, 0);
 
     return *this;
   }
 
   //-----------------------------------------------------------------------
-  u32 Buffer::get_resource_id() const
+  u32 Buffer::resource_id() const
   {
     return m_resource_id;
   }
   //-----------------------------------------------------------------------
-  u32 Buffer::get_resource_target() const
+  u32 Buffer::resource_target() const
   {
     return m_resource_target;
   }
 
   //-----------------------------------------------------------------------
   void Buffer::bind() const
   {
-#if REX_PLATFORM_X64
-    if (m_resource_target != GL_FRAMEBUFFER)
+    // When using OpenGL ES ( GLES ) the default framebuffer is not 0.
+    // When Emscripten is enabled OpenGL ES will always be used.
+#if __EMSCRIPTEN__
+    REX_ASSERT_X(m_resource_id != 0, "Binding an invalid resource id");
+#else
+    if(m_resource_target != GL_FRAMEBUFFER)
```

Comment:
I'd add the framebuffer check in the assert to avoid the overhead.
it'd be compiled out anyway, because the if body is empty, but still..